.golangci.yml

```diff
@@ -0,0 +1,31 @@
+linters:
+  disable-all: false                    # This enabled 7 default linters as of golangci-lint@v1.50.0:
+                                        #   1. errcheck
+                                        #   2. gosimple
+                                        #   3. govet
+                                        #   4. ineffassign
+                                        #   5. staticcheck
+                                        #   6. typecheck
+                                        #   7. unused
+  enable:
+    - gofmt
+    - whitespace
+    - gocognit
+    - unparam
+
+run:
+  timeout: 10m                          # golangci-lint run's timeout.
+
+  build-tags:                           # All linters will run with below mentioned build tags.
+    - "integration"
```

Comment:
@Bhargav-InfraCloud what is the significance of this build tag?

Reply:
It enables the linting on files tagged as `integration`.

Currently, the codebase consists of 4 build constraints in multiple files:

1. +build `!ignore_autogenerated`
2. +build `!integration`
3. +build `integration`
4. +build `!unit`

Among these, all but 3rd are negative build constraints and are fine by golangci-lint to do the scan. But 3rd stops golangci-lint from linting on files:

1. pkg/testing/e2e_test.go
2. pkg/testing/integration_register.go
3. pkg/testing/integration_test.go

Though these are all integration tests, minimal formatting linters like `gofmt` & `govet` are still required as per `goreportcard` results found on #1274.